src/main/java/mods/railcraft/common/util/inventory/InvTools.java

```diff
@@ -78,6 +78,34 @@ public static ItemStack emptyStack() {
         return null;
     }
 
+    /**
+     * This method changes the original stack!
+     */
+    @Nullable
+    public static ItemStack decSize(@Nullable ItemStack stack) {
```

Comment:
Mutating the stack is a bad idea generally. I've gotten tons of flack for doing that in the past from other modders who use object identity to detect changes to stacks.